### PR TITLE
[Infra] Add issue form for feature requests

### PR DESCRIPTION
**Pull request by ncooke3**
### Context

Related to #10119

[GitHub issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) offer a better way to format and template GitHub issues.

I have ported the markdown feature request template to the issue form syntax. I also added some fields that I think would be helpful for triage.

I have a fork where you can see it in action. 

See [here](https://github.com/ncooke3/firebase-ios-sdk/issues/new?assignees=&labels=&template=FEATURE_REQUEST.yml&title=%5BFR%5D%3A+) for what it looks like when opening an issue. 

And see [here](https://github.com/ncooke3/firebase-ios-sdk/issues/3) for what the created FR looks like.

#no-changelog

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          Firebase API and includes a brief API proposal that would make the use
+          case possible.
+
+        <!--- Links -->
+        [1]: https://github.com/firebase/firebase-ios-sdk/discussions/categories/pitches
+  - type: textarea
+    attributes:
+      label: Description
+      description: |
+        Describe the feature request. Explain the following:
```
Comment (paulb777) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:26]: wording nit: 
    
    Describe the feature request. Provide the following:
        1. Use case
        2. How the Firebase SDK _currently_ handles it
        3. How you propose the Firebase SDK _should_ handle it

**Review by paulb777 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml (hunk 1 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml]: Done in [5163007](https://github.com/firebase/firebase-ios-sdk/pull/10120/commits/516300786873a1c8f614911a329c4156618918dd)

**Review by chliangGoogle — APPROVED** [review]

**Review by rizafran — APPROVED** [review]
Looks good to me :)

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+name: ➕ Feature Request
+description: File a file request if you have a suggestion for a new feature.
+title: "[FR]: "
+labels: []
```
Comment (morganchen12) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:4]: add the feature request label?

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        - Analytics
+        - App Check
+        - App Distribution
+        - Authentication
+        - Crashlytics
+        - Database
+        - DynamicLinks
+        - Firestore
+        - Functions
+        - In App Messaging
```
Comment (morganchen12) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:60]: nit: `In-App` instead of `In App`

**Review by morganchen12 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml (hunk 2 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:4]: Good idea, thanks. Done.

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml (hunk 3 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:60]: Done.

**Pull request merged**